theano/gpuarray/dnn.py

```diff
@@ -2833,6 +2833,295 @@ def local_abstractconv3d_cudnn_graph(op, context_name, inputs, outputs):
     return [rval]
 
 
+class GpuDnnTransformerDescriptor(COp):
+
+    """
+    This Op builds a spatial transformer descriptor for use in spatial transformer network
+    operations.
+    """
+
+    __props__ = ('dtype',)
```

Comment:
Remark that dtype is not used at all in `make_node()`.

In fact, I suggest you totally remove `dtype` prop and param from all ops related to spatial transformation. Output dtype should simply be the same as input (img) dtype.